easybuild/tools/build_from_specsfile.py

```diff
@@ -0,0 +1,150 @@
+import yaml
+from easybuild.tools.robot import search_easyconfigs
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.base import fancylogger
+
+
+_log = fancylogger.getLogger('specsfile', fname=False)
+
+
+# general specs applicable to all commands
+class Specsfile(object):
+    def __init__(self):
+        self.easybuild_version = None
+        self.robot = False
+        self.software_list = []
+
+    # returns list of all commands - finished
+    def compose_full_paths(self):
+        easyconfigs_full_paths = []
+        for sw in self.software_list:
+            path_to_append = self.get_ec_path(sw)
+
+            if path_to_append is None:
+                continue
+            else:
+                easyconfigs_full_paths.append(path_to_append)
+        return easyconfigs_full_paths
+
+    # single command
+    def get_ec_path(self, sw):
+        full_path = search_easyconfigs(query='%s-%s-%s-%s' % (
+            str(sw.software), str(sw.version), str(sw.toolchain_name), str(sw.toolchain_version)),
+            short=False, filename_only=False,
+            terse=False, consider_extra_paths=True, print_result=False, case_sensitive=False)
+        if len(full_path) == 1:
+            return full_path[0]
+        else:
+            print('%s does not have clearly specified parameters - %s matches found. Skipping. \n' %
+                  (sw.software, str(len(full_path))))
+
+    # todo: flags applicable to all sw (i.e. robot)
+    def get_general_options(self):
+        general_options = {}
+        general_options['robot'] = self.robot
+        general_options['easybuild_version'] = self.easybuild_version
+        return general_options
+
+
+# single sw command
+class SoftwareSpecs(object):
+    def __init__(self, software, version, toolchain, toolchain_version, toolchain_name):
+        self.software = software
+        self.version = version
+        self.toolchain = toolchain
+        self.toolchain_version = toolchain_version
+        self.toolchain_name = toolchain_name
+        self.toolchain = toolchain
+
+        self.version_suffix = None
+
+    def get_version_suffix(self):
+        return self.version_suffix or ''
+
+
+# implement this to your own needs - to create custom yaml/json/xml parser
+class GenericSpecsParser(object):
```

Comment:
I would rename this to `EasyStackParser`, and assume we'll only support YAML format.

If we change our minds later, it should be easy to refactor to also support other formats...

"Premature optimizations is the root of all evil"